hail/src/main/scala/is/hail/expr/ir/functions/TableCalculateNewPartitions.scala

```diff
@@ -0,0 +1,26 @@
+package is.hail.expr.ir.functions
+
+import is.hail.expr.ir.{ExecuteContext, TableValue}
+import is.hail.expr.types.virtual._
+import is.hail.expr.types.TableType
+import is.hail.rvd.RVD
+import is.hail.utils._
+
+case class TableCalculateNewPartitions(
```

Comment:
I'd be OK with this being a full IR node - it's a great first step toward lowering TableKeyBy!